omnifit/spectrum/spectrum.py

```diff
@@ -4,765 +4,840 @@
+        Re-sort the data afterwards.
+
+        Parameters
+        ----------
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        self.convert2(u.kayser)
+
+    @clonable
+    def convert2wl(self):
+        """
+        convert2wl(clone=False)
+
+        Convert the x axis data to wavelength (in microns) units.
+        Re-sort the data afterwards.
+
+        Parameters
+        ----------
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        self.convert2(u.micron)
+
+    @clonable
+    def convert2(self, newunit):
+        """
+        convert2(newunit, clone=False)
+
+        Convert the x axis data to given spectral units.
+        Re-sort the data afterwards.
+
+        Parameters
+        ----------
+        newunit : `astropy.units.core.Unit`
+            Desired (spectral) unit the x axis data should be
+            converted to.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        with u.set_enabled_equivalencies(u.equivalencies.spectral()):
+            self.x = self.x.to(newunit)
+        self.__sort()
+
+    @clonable
+    def subspectrum(self, limit_lower, limit_upper):
+        """
+        subspectrum(limit_lower, limit_upper, clone=False)
+
+        Cropped the spectrum along along the x axis using the given
+        inclusive limits.
+
+        Parameters
+        ----------
+        limit_lower : `float`
+            The desired minimum x axis of the cropped spectrum, in
+            current units of the spectrum. This limit is inclusive.
+        limit_upper : `float`
+            The desired maximum x axis of the cropped spectrum, in
+            current units of the spectrum. This limit is inclusive.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        iSub = np.logical_and(
+            np.greater_equal(self.x.value, limit_lower),
+            np.less_equal(self.x.value, limit_upper))
+        newX = self.x[iSub]
+        newY = self.y[iSub]
+        self.x = newX
+        self.y = newY
+
+    @clonable
+    def interpolate(self, target_spectrum):
+        """
+        interpolate(target_spectrum,clone=False)
+
+        Interpolate spectrum to match target spectrum resolution.
+        Does not modify current spectrum, but returns a new one, which is
+        a copy of the current spectrum but with the interpolated data on
+        the x and y fields.
+        The target spectrum has to be using the same units on the x and
+        y axes as the current spectrum, or the interpolation will fail.
```

Comment:
```suggestion
        Does not modify current spectrum, but replaces it with a new one,
        which is a copy of the current spectrum but with the interpolated
        data on the x and y fields.
        The target spectrum has to be using the compatible units on the
        x and y axes as the current spectrum, or the interpolation will fail
        (including, e.g., units of wavenumbers/frequency/wavelength).
```
Update to reflect the wider acceptance of units; also “does not modify” sounded somewhat misleading, as one would rather expect that to work as `newspec = oldspec.interpolate(target_spectrum)`.

Reply:
Implemented in 275c50f